core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -106,7 +107,7 @@ public void onNoLongerMaster(String source) {
 
                 @Override
                 public void onFailure(String source, Exception e) {
-                    logger.error(new ParameterizedMessage("unexpected failure during [{}]", source), e);
+                    logger.error((Supplier<?>) () -> new ParameterizedMessage("unexpected failure during [{}]", source), e);
```

Comment:
I can't help but wonder if we shouldn't have a static helper function that does this for us?

```java
public static Supplier<ParameterizedMessage> message(String message, Object... params) {
  return new ParameterizedMessage(message, params);
}
````

It's of course another import, but it will probably be better than staring at `(Supplier<?>) () -> new ParameterizedMessage(...)` in so many places.